file_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"os"
+	"testing"
+	"path/filepath"
+	"io/ioutil"
+	"fmt"
+)
+
+// Although other tests besides those in this file require this env var, this init() func will cover all tests.
+func init() {
+	if os.Getenv("GITHUB_OAUTH_TOKEN") == "" {
+		fmt.Println("ERROR: These tests require that env var GITHUB_OAUTH_TOKEN be set to a GitHub Personal Access Token.")
+		fmt.Println("See the tests cases to see which GitHub repos the oAuth token needs access to.")
+		os.Exit(1)
```

Comment:
> When you call os.Exit(1), do any functions that were called with defer have a chance to run?

It's a great question. I just put some `println` statements in to see if they run and it looks like none of the tests execute until after `init()` is called.